tools/bump_version.sh

```diff
@@ -16,49 +21,107 @@ update_package_version() {
     sed -i "0,/${pattern}/s/${pattern}/version = \"${new_version}\"/1" $1
 }
 
-# Update Docker image versions (`asterinas/asterinas:{version}`) in file $1
+# Updates Docker image versions (`asterinas/asterinas:{version}`) in file $1
 update_image_versions() {
     echo "Updating file $1"
     sed -i "s/asterinas\/asterinas:[[:digit:]]\+\.[[:digit:]]\+\.[[:digit:]]\+/asterinas\/asterinas:${new_version}/g" $1
 }
 
+# Prints the help message
+print_help() {
+  echo "Usage: $0 [increment_type]"
+  echo "Increment_type can be one of: patch, minor, or major. If not provided, the default is 'minor'."
+}
+
+# Adds the number by 1
+# Bash cannot deal with 0 by using `$((num++))`,
+# So I add a function here to specially deal with 0.
+add_one() {
+  local num=$1
+  if [ "$num" == "0" ]; then
+    echo "1"
+  else
+    local incremented=$((num + 1))
+    echo "$incremented"
+  fi
+}
+
+# Increases the version based on $increment_type
+increment_version() {
+  local IFS="."
+  local version_parts=($current_version)
+
+  case "$increment_type" in
+    "patch")
+      version_parts[2]=$(add_one "${version_parts[2]}")
+      ;;
+    "minor")
+      version_parts[1]=$(add_one "${version_parts[1]}")
+      version_parts[2]=0
+      ;;
+    "major")
+      version_parts[0]=$(add_one "${version_parts[0]}")
+      version_parts[1]=0
+      version_parts[2]=0
+      ;;
+  esac
+
+  echo "${version_parts[*]}"
+}
+
+# Validates the increment type
+validate_increment_type() {
+  case "$increment_type" in
+    "patch" | "minor" | "major")
+      ;;
+    *)
+      echo "Error: Invalid increment_type. Allowed values are: patch, minor, or major."
+      print_help
+      exit 1
+      ;;
+  esac
+}
+
 SCRIPT_DIR=$( cd "$( dirname "${BASH_SOURCE[0]}" )" && pwd )
 ASTER_SRC_DIR=${SCRIPT_DIR}/..
 CARGO_TOML_PATH=${ASTER_SRC_DIR}/Cargo.toml
 OSDK_CARGO_TOML_PATH=${ASTER_SRC_DIR}/osdk/Cargo.toml
 VERSION_PATH=${ASTER_SRC_DIR}/VERSION
 
-# Get and check the new version number
-if [[ $1 =~ ^[0-9]+\.[0-9]+\.[0-9]+$ ]]; then
-    new_version=$1
-else
-    printf "Invalid version number: $1\nUsage: ./tools/bump_version.sh <new_version>\n"
-    exit -1
+current_version=$(cat ${VERSION_PATH})
+increment_type=${1:-"minor"}
+
+if [[ "$increment_type" == "--help" || "$increment_type" == "-h" ]]; then
+  print_help
+  exit 0
 fi
 
-# Update the package version in Cargo.toml
+validate_increment_type
+new_version=$(increment_version ${current_version})
+
+# Updates the package version in Cargo.toml
 update_package_version ${CARGO_TOML_PATH}
 update_package_version ${OSDK_CARGO_TOML_PATH}
 
-# Automatically bump Cargo.lock file
+# Automatically bumps Cargo.lock file
 cargo update -p asterinas --precise $new_version
 
-# Update Docker image versions in README files
+# Updates Docker image versions in README files
```

Comment:
Please discard changes like "Update --> Updates".

Yes, when documenting a Rust method, we want to use the third person singular form of the starting verb such as "Updates". This is because the Rust doc should be read as "The method updates ...".

But here the comments are written for the code inside a script or function. In such cases, the comments should be read as "We update ..." or "We ask the shell interpreter to update ..." or "We ask the CPU to update ...". So the verb should be in the first person plural form such as "Update".